api/utils/keys/ecdsa.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package keys
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+
+	"github.com/gravitational/trace"
+	"golang.org/x/crypto/ssh"
+)
+
+// ECDSAPrivateKey is an ecdsa.PrivateKey with additional methods.
+type ECDSAPrivateKey struct {
+	*ecdsa.PrivateKey
+	privateKeyDER []byte
+	sshPub        ssh.PublicKey
+}
+
+// newECDSAPrivateKey creates a new ECDSAPrivateKey from a ecdsa.PrivateKey.
+func newECDSAPrivateKey(priv *ecdsa.PrivateKey) (*ECDSAPrivateKey, error) {
+	keyDER, err := x509.MarshalECPrivateKey(priv)
```

Comment:
Any specific reason why we're going for SEC 1 and PKCS#1 here and in rsa.go, rather than always using PKCS#8?

Reply:
No reason other than maintaining old behavior. But I think you're right, we might as well upgrade to PKCS#8 here.